### ISSUE TITLE
InsertEntryAsync - Value cannot be null. Parameter name: source

### ISSUE THREAD

**Issue by NoMercy** [object/Simple.OData.Client#157]
Hello,

I am not able to fix one problem.
When I am inserting (just one model, others from Edm are working) I get exception message which is not descriptive. It is:
System.ArgumentNullException: Value cannot be null. Parameter name: source

It is falling somewhere in InsertEntryAsync function.

I see in fiddler, that request on metadata was sent and it is all. It is probably falling when the insert request is created.

Insert in Postman works:
{
  "@odata.context":"https://Endpoint/$metadata#TaskAssignmentRules/$entity",
"DelayMinutes":900,"IsByPreference":true,"Position":1,
"PreferenceLevel":2,"TaskId":1032543
}

The second think I observated is that I am able to insert with null collections from Postman. It is not possible with Simple.OData (I must create empty collection). It is not important now.

It is not all code but maybe it could help.

![image](https://cloud.githubusercontent.com/assets/121787/9872339/08333c5c-5b99-11e5-9146-a210e8e7c760.png)

    public class TaskAssignmentRuleViewModel
    {
        [JsonProperty("DelayMinutes")]
        public int DelayMinutes { get; set; }

        [JsonProperty("ErrorMessage")]
        public string ErrorMessage { get; set; }

        [JsonProperty("ExecutedOn")]
        public DateTimeOffset? ExecutedOn { get; set; }

        [JsonProperty("Id")]
        public int Id { get; set; }

        [JsonProperty("IsByPreference")]
        public bool IsByPreference { get; set; }

        [JsonProperty("Position")]
        public int Position { get; set; }

        [JsonProperty("PreferenceLevel")]
        public int? PreferenceLevel { get; set; }

        [JsonProperty("TaskId")]
        public int TaskId { get; set; }

        [JsonProperty("TaskOrderDate")]
        public DateTimeOffset? TaskOrderDate { get; set; }

        [JsonProperty("AssigneeIds")]
        public IList<int> AssigneeIds { get; set; }

        /// <summary>
        /// Tags that will be used for assignees selection.
        /// </summary>
        [JsonProperty("Tags")]
        public IList<int> Tags { get; set; }

    }

I tried to debug it and it is better when it is not asynchronous. But still the same problem.

// Create new object
var newRule = new TaskAssignmentRuleViewModel
{
  DelayMinutes = 90,
  IsByPreference = true,
  Position = 1,
  PreferenceLevel = 2,
  TaskId = taskId,
  AssigneeIds = emptyList
};

var c = clientBeta.For<TaskAssignmentRuleViewModel>();
// var c = clientBeta.For("TaskAssignmentRules"); 
var s = c.Set(newRule);
// Insert new rule
var ruleId = s.InsertEntryAsync(true, cancellationToken).Result;

I was using version 4.8.4. I am using 4.10 now.
Can you check it, please?
Thank you.

**Comment by NoMercy** [comment]
Solved with HttpClient (it is a simple request) so don't hurry.
I am looking forward to your answer. 
Have a nice day.

**Comment by sixlettervariables** [comment]
If you assign `Tags` an empty collection (as you did with `AssigneeIds`) does the exception go away?

**Comment by NoMercy** [comment]
It's the same. I tried empty list, list of IDs.... didn't help.

**Comment by object** [comment]
I will investigate this further, thank you for reporting it.

**Comment by object** [comment]
Well, it really took a while :-) However, I inspected your metadata and code and found 2 things:
1. Metadata had property "DelayMinues" while the C# entity had "DelayMinutes".
2. Tags property was not set, and that caused an exception when generating POST payload. Once I set Tags to an empty list (like AssigneeIds) the request was generated and sent to the mock server.
Sorry for leaving this one unanswered for so long.
